core/io/marshalls.cpp

```diff
@@ -1814,6 +1814,82 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 	return OK;
 }
 
+Error to_raw_bytes(const Variant &p_variant, PackedByteArray &out) {
+	switch (p_variant.get_type()) {
+		case Variant::INT: {
+			int32_t v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+		} break;
+		case Variant::BOOL: {
+			bool v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+		} break;
+		case Variant::FLOAT: {
+#ifdef REAL_T_IS_DOUBLE
+			double v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+#else
+			float v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+#endif
```

Comment:
This is not correct, `float` in `Variant` is always 64 bits, only the values of `Vector2/3/4` are 32/64 bits, this needs to be changed

Reply:
float is 32 bits and only 64 bits when REAL_T_IS_DOUBLE is defined.. Variant, does not matter as there is no mapping from Variant to GLSL types. Vectors in GLSL are 16 bytes aligned and Vector2 for example will take 2xfloat - 8 bytes and 8 pad bytes. Padding is not the part of the job for this function.

Reply:
Float in GDScript is 64 bits

Reply:
Removed use of the "double" completely, added semantic conversion to 32 bit float at a cost of precision.